### PR TITLE
Add configured systems link to configuration profiles

### PR DESCRIPTION
**Pull request by epostea**
- Updated the virtual machines collector to retrieve the virtual machines with the parent stack details. The stack includes the `templateId` that maps to our configuration profile in manageiq.

### CODE REVIEW

File: app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
+    configuration_profiles = collector.templates.map do |template|
       persister.configuration_profiles.build(
         :manager_ref     => template["id"].to_s,
         :name            => template["name"],
         :description     => template["description"],
         :target_platform => template.dig("manifest", "template_provider")
       )
     end
+
+    configuration_profiles
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb:17]: `collector.templates.map` will automatically return the array so you can skip the local variable plus the return and just do

```ruby
def configuration_profiles
  configuration_profiles = collector.templates.map do |template|
    ...
  end
end
```

File: app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb [hunk 2]
```diff
@@ -33,23 +35,33 @@ def configuration_profiles
...
   #     "tenantId": "fbb9ab6a-b54d-43d4-9200-6d950700588f",
   #     "namespaceId": "acme",
   #     "namespaceMapping": true,
   #     "id": "5ee8ddde6143d40017c2ee45"
   # }]
-  def configured_systems
+  def configured_systems(persister_profiles)
+    persister_profiles_by_ref = persister_profiles.index_by do |profile|
+      profile.data[:manager_ref]
+    end
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb:50]: If you're indexing by the manager_ref and you have all the data needed then you could just do a lazy_find instead

File: app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb [hunk 3]
```diff
@@ -33,23 +35,33 @@ def configuration_profiles
...
+
     collector.virtual_machines.each do |virtual_machine|
-      virtual_instance_ref = virtual_machine["idFromProvider"]
-      counterpart          = persister.vms.lazy_find(virtual_instance_ref) if virtual_instance_ref
+      virtual_instance_ref  = virtual_machine["idFromProvider"]
+      counterpart           = persister.vms.lazy_find(virtual_instance_ref) if virtual_instance_ref
+      template_id           = virtual_machine.dig("stacks", "templateId")
+      configuration_profile = persister_profiles_by_ref[template_id.to_s] if template_id
```
Comment (agrare) [inline app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb:56]: so e.g.
```
configuration_profile = persister.configuration_profiles.lazy_find(template_id.to_s) if template_id
...

Comment (epostea) [reply app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb]: Thanks, the code is much cleaner after your suggestions :-)

**Review by agrare — APPROVED** [review]
:+1: looks awesome

**Pull request merged**